fuzzbench/conftest.py

```diff
@@ -11,25 +11,13 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""Fake jobs."""
+"""A pytest conftest.py file that defines fixtures."""
+import redis
 
-import subprocess
-import time
+import pytest
 
 
-def build_image(name):
-    """Build a Docker image."""
-    print('Building', name)
-    subprocess.run(['docker', '--version'], check=True)
-    time.sleep(3)
-    return True
-
-
-def run_trial():
-    """Run a trial."""
-    return True
-
-
-def measure_corpus_snapshot():
-    """Measure a corpus snapshot."""
-    return True
+@pytest.fixture
+def redis_connection():
+    """Returns the default redis server connection."""
+    return redis.Redis(host='queue-server')
```

Comment:
This works? It doesn't need to be an IP? 
I guess the device you are running this on knows an ip that queue-server maps to?

Reply:
>This works? 

Yes
>It doesn't need to be an IP?

I think here it matches the pattern in fuzzbench/run_experiment.py

https://github.com/google/fuzzbench/blob/5db871f65c312d4b2cedf16bd7ac2c6311987b26/fuzzbench/run_experiment.py#L42-L46

>I guess the device you are running this on knows an ip that queue-server maps to?

I think it uses the docker container here:
https://github.com/google/fuzzbench/blob/5db871f65c312d4b2cedf16bd7ac2c6311987b26/compose/fuzzbench.yaml#L26-L27

Reply:
do we need a separate conftest.py file for this?

can we put this fixture into test_e2e_run.py?

can we make it's scope modul / class level (ie. run it only once before calling the test_... methods)?

Reply:
> do we need a separate conftest.py file for this?

I follow the pattern of current code. We will have specific conftest.py in each folder.

> can we put this fixture into test_e2e_run.py?

I guess put it aside will be better and clearer? wdyt? @jonathanmetzman 

> can we make it's scope modul / class level (ie. run it only once before calling the test_... methods)?

Shall we do that @jonathanmetzman ?

Reply:
I don't really think this is an intended pattern. If a fixture will be used in multiple files then put it in conftest.py otherwise can go in the file it is used.

Reply:
Yes. So we already put it into the e2e-test since this fixture is only used here now. Done.